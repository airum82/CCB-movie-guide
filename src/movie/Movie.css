.movie {
  background-color: var(--color4);
  height: 100%;
  width: 100%;
}

.movie h2 {
  text-align: center;
  font-size: 100%;
  height: 7vh;
}

.movie img {
  height: 50vh;
  display: block;
  margin: 0 auto;
  margin-bottom: 4%;
}

.movie:hover {
  color: white;
  cursor: pointer;
}


.movie-single h2 {
  text-align: center;
}

.movie-single img {
  display: block;
  height: 90vh;
  margin: 2% auto;
}

.movie-description {
  width: 65%;
  margin: 0 auto;
}

.movie-stats {
  text-align: center;
}

@media only screen and (max-width: 850px) {
  .movie img {
    height: 40vh;
  }
}

@media only screen and (max-width: 450px) {
  .movie {
    width: 90%;
    margin: 0 auto;
    height: 83%;
  }

  .movie img {
    height: 70vh;
  }
}

@media only screen and (max-width: 530px) {
  .movie-description {
    width: 90%;
  }
}