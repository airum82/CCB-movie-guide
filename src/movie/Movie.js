import React from 'react';
import PropTypes from 'prop-types';
import './Movie.css';

const Movie = ({ movie, formatDate, viewMovie, location }) => {
  const singleView = location.pathname === `/movie/${movie.id}`;
  return (
      <article 
        className={ singleView ? 'movie-single' : 'movie'} 
        onClick={() => {
          viewMovie(movie.id)
        }}
      >
        <h2>{movie.title}</h2>
        <img src={`https://image.tmdb.org/t/p/w500${movie.poster_path}`} />
        { singleView ?
          <div className="movie-details">
            <p className="movie-description">{movie.overview}</p>
            <p className="movie-stats">Release Date: {formatDate(movie.release_date)}</p>
            <p className="movie-stats">Voter Average: {movie.vote_average}</p>
          </div> : ''
        
           }
      </article>
  )
}

Movie.propTypes = {
  movie: PropTypes.object,
  formatDate: PropTypes.func,
  viewMovie: PropTypes.func,
  location: PropTypes.object
}

export default Movie;