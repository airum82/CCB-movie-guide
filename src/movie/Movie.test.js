import React from 'react';
import Movie from './Movie';
import { shallow } from 'enzyme';

describe('Movie', () => {
  let wrapper;
  const mockProps = {
    movie: {
      id: '583',
      title: 'the fakening',
      overview: 'so so fake',
      release_date: '1-35-3000',
      vote_average: '10',
      poster_path: 'thispath/3445'
    },
    formatReleaseDate: jest.fn(),
    viewMovie: jest.fn(),
    location: {
      pathname: '/now_playing'
    }
  }
  beforeEach(() => {
    wrapper = shallow(<Movie {...mockProps} />)
  })

  it('should match snapshot upon render', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render movie details when pathname is "movie/id"', () => {
    mockProps.location.pathname = '/movie/583';
    wrapper = shallow(<Movie {...mockProps} />);
    expect(wrapper.find('.movie-details').length).toBe(1);
  })

  it('should not render movie details when pathname is not "movie/id"', () => {
    mockProps.location.pathname = '/now_playing'
    wrapper = shallow(<Movie {...mockProps} />)
    expect(wrapper.find('.movie-details').length).toBe(0)
  })

  //prop functions
  it('should call viewMovie with correct params when clicked on', () => {
    wrapper.find('article').simulate('click');
    expect(mockProps.viewMovie).toHaveBeenCalledWith(mockProps.movie.id)
  })

  it('should call formatReleaseDate with correct params on render', () => {
    mockProps.location.pathname = '/movie/583';
    wrapper = shallow(<Movie {...mockProps} />);
    expect(mockProps.formatReleaseDate).toHaveBeenCalledWith(mockProps.movie.release_date);
  })
})