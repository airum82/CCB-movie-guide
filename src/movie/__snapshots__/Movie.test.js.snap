// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match snapshot upon render 1`] = `
<article
  className="movie"
  onClick={[Function]}
>
  <h2>
    the fakening
  </h2>
  <img
    src="https://image.tmdb.org/t/p/w500thispath/3445"
  />
</article>
`;
